base/interactiveutil.jl

```diff
@@ -64,7 +64,23 @@ function edit(path::AbstractString, line::Integer=0)
     nothing
 end
 
-edit(f)          = edit(functionloc(f)...)
+function edit(f)
+    ms = methods(f)
+    nm = length(ms)
+    if nm <= 1
+        edit(functionloc(f)...)
+    else
+        for (i, m) in enumerate(ms)
+          @printf("%4u: %s\n", i, m)
+        end
+        print("Which method? [1-$nm] (<Enter> to cancel) > ")
+        (ix_str = strip(readline())) == "" && return
+        ix = parse(Int, ix_str)
+        @assert ix >= 1 && ix <= nm "$ix ∉ [1-$nm]"
```

Comment:
The explicit check instead of a boundserror is good, just that `@assert` should only be used for internal consistency check and not for user errors. I'm not sure what's the best kind of error here, maybe a `error` or simply retry.